### DISCUSSION TITLE
How to achieve a rich appearance for the dungeon

### DISCUSSION THREAD

**Discussion by shun126** [shun126/UE5-DungeonGeneratorDemo#40] [Q&A]
If the artist can control the method of selecting floor,roof,wall meshes, would it be possible to generate richer dungeons?
I’d like to know what the artists keep in mind when creating the modules.

How to connect meshes and place interior decorations. I need some tips.

**Comment by shun126** [comment]
![20240809](https://github.com/user-attachments/assets/0dfc7377-32de-471e-a438-75e5baa63ead)
I tried out a system for generating foliage. I scattered grass and small stones along the walls.
Would organizing this system be beneficial for artists?

https://x.com/happy_game_dev/status/1821587859566133456?s=46&t=GRC1VbwXwpi2NsxlnZrhEg

**Reply by shun126** [reply]
![GUzB50ebkAAYjYv](https://github.com/user-attachments/assets/9b0fb073-9b38-4b59-86e1-d65faef05d90)
I tested a system for generating foliage. Gravel and moss seem more suitable than grass. I’m also considering whether I can use the same system for ivy or decorations on walls and ceilings. I’m currently uncertain about the collision filter to use when attaching foliage to terrain. It might be more user-friendly if channels can be specified, rather than just fixing it to Pawn...

I want to create an explanation video, but the avatar system is currently stuck, making it difficult to make progress or go back. I’ll keep working hard.

**Reply by shun126** [reply]
![image](https://github.com/user-attachments/assets/494515d9-98b8-4fdf-8881-ea51b993d82d)
![image](https://github.com/user-attachments/assets/90eb33aa-c521-451d-ac7e-51d15914fa80)

I used AI image generation to visualize the results of the dungeon generator. The vending machines could be implemented through interior generation. By switching meshes, posters on the walls could be added, and with the planned foliage extension, small items could be placed on the floor. Now, I just need a method to implement ceiling air conditioning units.

**Comment by obsidiousdev** [comment]
Air Conditioning units worked in my dungeons by spawning blueprints in the middle of rooms, and when spawned each unit would cast outwards and check for how many tiles the dungeon generated in the room and extending the meshes out accordingly.
![image](https://github.com/user-attachments/assets/ff71301f-26f5-4525-af51-84633e63cb1a)
![image](https://github.com/user-attachments/assets/6c21558a-9e83-4015-aa06-db70b05bb6ec)

**Reply by shun126** [reply]
I see! That’s a good approach.


I should have given more thought to the design of the ceiling and walls.

**Comment by shun126** [comment]
The model was created by yamaguchi◆3DCG ( https://coconala.com/users/4493115 ).
I feel like adjusting the lighting has brought out a great atmosphere.
If I can add a mesh selection feature via Blueprint in addition to sublevels, I think it will allow for even richer expressions.

パーツのモデルをyamaguchi◆3DCG（ https://coconala.com/users/4493115 ） 様に制作していただきました。
あわせてライティングを調整した事で空気感が出てきた気がします。
サブレベルに加えてBlueprintによるメッシュの選択機能追加ができると、より豊かな表現ができそうです。

![Unreal Editor 2024_11_24 15_09_34](https://github.com/user-attachments/assets/37c42d79-d0f0-4750-a6f5-7ae9ec2659c6)
